pkg/controller/servicebindingrequest/bind_custom_resources.go

```diff
@@ -0,0 +1,104 @@
+package servicebindingrequest
+
+import (
+	"github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+var (
+	path = map[string][][]string{
+		"ConfigMap": {{"data"}},
+		"Secret":    {{"data"}},
+		"Route":     {{"spec", "host"}},
+		"Service":   {{"spec", "clusterIP"}},
+	}
+)
+
+// BindNonBindableResources struct contains information about operator backed CR and
+// list of expected GVRs to extract information from.
+type BindNonBindableResources struct {
+	sbr              *v1alpha1.ServiceBindingRequest
+	cr               *unstructured.Unstructured
+	resourcesToCheck []schema.GroupVersionResource
+	client           dynamic.Interface
+	data             map[string]interface{}
+}
+
+// NewBindNonBindable returns new instance
+func NewBindNonBindable(
+	sbr *v1alpha1.ServiceBindingRequest,
```

Comment:
`sbr` is not used anywhere else in this file other than setting `BindNonBindableResources` field; dependency can be removed.

Reply:
removed @isutton